### ISSUE TITLE
`xpra initialization error:  connection failed: 'NoneType' object has no attribute 'getsockname'`

### ISSUE THREAD

**Issue by stdedos** [Xpra-org/xpra#3450]
**Describe the bug**
I am in a shadow-reconnect loop (for what reason, I don't know).

Among all the initialization text from the client, I am getting:
```py
2022-01-27 12:22:02,924 Xpra GTK3 X11 client version 4.4-r30771 (g636b0b8d8) 64-bit
2022-01-27 12:22:02,999  running on Linux Ubuntu 20.04 focal
2022-01-27 12:22:03,000  window manager is 'GNOME Shell'
2022-01-27 12:22:03,624 GStreamer version 1.16.2 for Python 3.8.10 64-bit
2022-01-27 12:22:03,648 created unix domain socket '/run/user/1000/xpra/clients/stdedos-Precision-3560-54867'
2022-01-27 12:22:03,695 No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
2022-01-27 12:22:03,805 Warning: vendor 'Intel' is greylisted,
2022-01-27 12:22:03,806  you may want to turn off OpenGL if you encounter bugs
2022-01-27 12:22:03,864 OpenGL enabled with Mesa Intel(R) Xe Graphics (TGL GT2)
xpra initialization error:
 connection failed: 'NoneType' object has no attribute 'getsockname'
```

Among that, there's also
<pre><font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ tree 
<font color="#3465A4"><b>.</b></font>
├── <font color="#3465A4"><b>0</b></font>
│   └── <font color="#3465A4"><b>tmp</b></font>
│       ├── <font color="#75507B"><b>xpra-notification-icon-ni3u44ct.png</b></font>
│       └── <font color="#75507B"><b>xpra-notification-icon-s5zis7wm.png</b></font>
├── <font color="#3465A4"><b>clients</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-12424</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54230</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54290</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54339</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54377</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54417</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54469</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54511</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54571</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54635</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54779</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54867</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54931</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-54983</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55039</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55079</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55122</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55166</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55213</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55278</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55327</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55369</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55421</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55463</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55513</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55568</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55613</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55659</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55890</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55948</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-55996</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56052</b></font>
│   └── <font color="#75507B"><b>stdedos-Precision-3560-7853</b></font>
├── <font color="#4E9A06"><b>run-xpra</b></font>
└── <font color="#3465A4"><b>tmp</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-0_pdimpb.png</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-7ira0cvi.png</b></font>
    └── <font color="#75507B"><b>xpra-notification-icon-96z7m63w.png</b></font>

4 directories, 39 files
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ xpra list
Found the following xpra sessions:
/run/user/1000/xpra/clients:
	UNKNOWN session at :12424
	UNKNOWN session at :54230
	UNKNOWN session at :54290
	UNKNOWN session at :54339
	UNKNOWN session at :54377
	UNKNOWN session at :54417
	UNKNOWN session at :54469
	UNKNOWN session at :54511
	UNKNOWN session at :54571
	UNKNOWN session at :54635
	UNKNOWN session at :54779
	UNKNOWN session at :54867
	UNKNOWN session at :54931
	UNKNOWN session at :54983
	UNKNOWN session at :55039
	UNKNOWN session at :55079
	UNKNOWN session at :55122
	UNKNOWN session at :55166
	UNKNOWN session at :55213
	UNKNOWN session at :55278
	UNKNOWN session at :55327
	UNKNOWN session at :55369
	UNKNOWN session at :55421
	UNKNOWN session at :55463
	UNKNOWN session at :55513
	UNKNOWN session at :55568
	UNKNOWN session at :55613
	UNKNOWN session at :55659
	UNKNOWN session at :55890
	UNKNOWN session at :55948
	UNKNOWN session at :55996
	UNKNOWN session at :56052
	UNKNOWN session at :7853
Re-probing unknown sessions in: /run/user/1000/xpra/clients
	UNKNOWN session at :12424 (cleaned up)
	UNKNOWN session at :54230 (cleaned up)
	UNKNOWN session at :54290 (cleaned up)
	UNKNOWN session at :54339 (cleaned up)
	UNKNOWN session at :54377 (cleaned up)
	UNKNOWN session at :54417 (cleaned up)
	UNKNOWN session at :54469 (cleaned up)
	UNKNOWN session at :54511 (cleaned up)
	UNKNOWN session at :54571 (cleaned up)
	UNKNOWN session at :54635 (cleaned up)
	UNKNOWN session at :54779 (cleaned up)
	UNKNOWN session at :54867 (cleaned up)
	UNKNOWN session at :54931 (cleaned up)
	UNKNOWN session at :54983 (cleaned up)
	UNKNOWN session at :55039 (cleaned up)
	UNKNOWN session at :55079 (cleaned up)
	UNKNOWN session at :55122 (cleaned up)
	UNKNOWN session at :55166 (cleaned up)
	UNKNOWN session at :55213 (cleaned up)
	UNKNOWN session at :55278 (cleaned up)
	UNKNOWN session at :55327 (cleaned up)
	UNKNOWN session at :55369 (cleaned up)
	UNKNOWN session at :55421 (cleaned up)
	UNKNOWN session at :55463 (cleaned up)
	UNKNOWN session at :55513 (cleaned up)
	UNKNOWN session at :55568 (cleaned up)
	UNKNOWN session at :55613 (cleaned up)
	UNKNOWN session at :55659 (cleaned up)
	UNKNOWN session at :55890 (cleaned up)
	UNKNOWN session at :55948 (cleaned up)
	UNKNOWN session at :55996 (cleaned up)
	UNKNOWN session at :56052 (cleaned up)
	UNKNOWN session at :7853 (cleaned up)
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ tree 
<font color="#3465A4"><b>.</b></font>
├── <font color="#3465A4"><b>0</b></font>
│   └── <font color="#3465A4"><b>tmp</b></font>
│       ├── <font color="#75507B"><b>xpra-notification-icon-ni3u44ct.png</b></font>
│       └── <font color="#75507B"><b>xpra-notification-icon-s5zis7wm.png</b></font>
├── <font color="#3465A4"><b>clients</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56106</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56150</b></font>
│   └── <font color="#75507B"><b>stdedos-Precision-3560-56195</b></font>
├── <font color="#4E9A06"><b>run-xpra</b></font>
└── <font color="#3465A4"><b>tmp</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-0_pdimpb.png</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-7ira0cvi.png</b></font>
    └── <font color="#75507B"><b>xpra-notification-icon-96z7m63w.png</b></font>

4 directories, 9 files
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ tree 
<font color="#3465A4"><b>.</b></font>
├── <font color="#3465A4"><b>0</b></font>
│   └── <font color="#3465A4"><b>tmp</b></font>
│       ├── <font color="#75507B"><b>xpra-notification-icon-ni3u44ct.png</b></font>
│       └── <font color="#75507B"><b>xpra-notification-icon-s5zis7wm.png</b></font>
├── <font color="#3465A4"><b>clients</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56106</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56150</b></font>
│   └── <font color="#75507B"><b>stdedos-Precision-3560-56195</b></font>
├── <font color="#4E9A06"><b>run-xpra</b></font>
└── <font color="#3465A4"><b>tmp</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-0_pdimpb.png</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-7ira0cvi.png</b></font>
    └── <font color="#75507B"><b>xpra-notification-icon-96z7m63w.png</b></font>

4 directories, 9 files
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ tree 
<font color="#3465A4"><b>.</b></font>
├── <font color="#3465A4"><b>0</b></font>
│   └── <font color="#3465A4"><b>tmp</b></font>
│       ├── <font color="#75507B"><b>xpra-notification-icon-ni3u44ct.png</b></font>
│       └── <font color="#75507B"><b>xpra-notification-icon-s5zis7wm.png</b></font>
├── <font color="#3465A4"><b>clients</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56106</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56150</b></font>
│   └── <font color="#75507B"><b>stdedos-Precision-3560-56195</b></font>
├── <font color="#4E9A06"><b>run-xpra</b></font>
└── <font color="#3465A4"><b>tmp</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-0_pdimpb.png</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-7ira0cvi.png</b></font>
    └── <font color="#75507B"><b>xpra-notification-icon-96z7m63w.png</b></font>

4 directories, 9 files
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ tree
<font color="#3465A4"><b>.</b></font>
├── <font color="#3465A4"><b>0</b></font>
│   └── <font color="#3465A4"><b>tmp</b></font>
│       ├── <font color="#75507B"><b>xpra-notification-icon-ni3u44ct.png</b></font>
│       └── <font color="#75507B"><b>xpra-notification-icon-s5zis7wm.png</b></font>
├── <font color="#3465A4"><b>clients</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56106</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56150</b></font>
│   ├── <font color="#75507B"><b>stdedos-Precision-3560-56195</b></font>
│   └── <font color="#75507B"><b>stdedos-Precision-3560-56249</b></font>
├── <font color="#4E9A06"><b>run-xpra</b></font>
└── <font color="#3465A4"><b>tmp</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-0_pdimpb.png</b></font>
    ├── <font color="#75507B"><b>xpra-notification-icon-7ira0cvi.png</b></font>
    └── <font color="#75507B"><b>xpra-notification-icon-96z7m63w.png</b></font>

4 directories, 10 files
<font color="#4E9A06"><b>stdedos@stdedos-Precision-3560</b></font>:<font color="#3465A4"><b>/run/user/1000/xpra</b></font>$ xpra list
Found the following xpra sessions:
/run/user/1000/xpra/clients:
	UNKNOWN session at :56106
	UNKNOWN session at :56150
	UNKNOWN session at :56195
	UNKNOWN session at :56249
Re-probing unknown sessions in: /run/user/1000/xpra/clients
	UNKNOWN session at :56106 (cleaned up)
	UNKNOWN session at :56150 (cleaned up)
	UNKNOWN session at :56195 (cleaned up)
	UNKNOWN session at :56249 (cleaned up)
</pre>

I don't know if the errors are related or not, but the `exec` of the client does seem to cause the above leaks.

**To Reproduce**
1. `xpra shadow ssh://stdedos@192.168.0.73/:0`
2. ~~client command~~
3. specific action to trigger the bug _No idea_

**System Information (please complete the following information):**
 - Server/Client OS: Ubuntu Focal
 - Xpra Server/Client Version [e.g. 4.2.3]: 4.4-r30771 (g 636b0b8d8) 

**Additional context**
Please see "reporting bugs" in the wiki section.

**Comment by stdedos** [comment]
Constant re-launching was my machinery (`while ! xpra shadow ssh://stdedos@192.168.0.73/:0 ; do sleep 5 ; done`), sorry about that.

However, "with that kind of exit", there definitely seems to be some resource leak

**Comment by totaam** [comment]
I hit this issue whilst testing and fixed it: 2cd9d6a2a28e20605de3b177a1be9acfca2db3cd

The error code should be set when this type of connection fails:
https://github.com/Xpra-org/xpra/blob/8e621dd2f2ea66b741b1b6f4849e92133f61d2a2/xpra/exit_codes.py#L14
